python/ml4ir/applications/ranking/tests/test_masking.py

```diff
@@ -83,4 +83,54 @@ def test_mask_all_query_feature(self):
         condition = tf.reduce_all(tf.logical_or(tf.equal(total_sum_per_batch, 5), tf.equal(total_sum_per_batch, 0)))
         self.assertTrue(condition.numpy(), True)
 
+    def test_create_fixed_masks(self):
+        mask = QueryFeatureMask(name="query_feature_mask",
+                                mask_rate=0,
+                                mask_at_inference=True,
+                                requires_mask=True)
+        feature_dim = 2
+        expected_result = [(0, 0), (0, 1), (1, 0), (1, 1)]
+        result = mask.create_fixed_masks(feature_dim)
+        self.assertEqual(result, expected_result)
+
+        feature_dim = 3
+        expected_result = [
+            (0, 0, 0), (0, 0, 1), (0, 1, 0), (0, 1, 1),
```

Comment:
@mohazahran note that this also creates mask with all 0s. It might be undesirable.

Reply:
for future - We could accept a range of values from user on how many they want in each mask.

So if there are 10 features and user wants [2, 4] features in each mask, then we do 10C2 + 10C3 + 10C4 masks.

Reply:
I addressed these comments in the thread above.